### PR TITLE
tty-ask-password: let's copy argv[] before forking

### PR DESCRIPTION
**Pull request by poettering**
This is an alternative fix for #12180.

### CODE REVIEW

File: src/tty-ask-password-agent/tty-ask-password-agent.c [hunk 1]
```diff
@@ -715,18 +720,24 @@ static int ask_on_this_console(const char *tty, pid_t *ret_pid, int argc, char *
...
-                                break;
+                STRV_FOREACH(i, arguments) {
+                        char *k;
+
+                        if (!streq(*i, "--console"))
+                                continue;
+
+                        k = strjoin("--console=", tty);
+                        if (!k) {
+                                log_oom();
```
Comment (keszybz) [inline src/tty-ask-password-agent/tty-ask-password-agent.c:732]: I think it'd be reasonable to do this calculation before forking, to simplify error handling and reporting.

**Comment by keszybz** [comment]
LGTM.

**Comment by yuwata** [comment]
Thanks.

**Comment by Keruspe** [comment]
This doesn't fix #12180 (or something else broke it since then.

Now I get:

```
(sd-password) takes no arguments
systemd-tty-ask-password-agent: Password agent failed with: 1
```

Different problem, same symptoms, while the other PR from @yuwata really did fix the symptoms

**Comment by poettering** [comment]
> Different problem, same symptoms, while the other PR from @yuwata really did fix the symptoms

Urks I fucked that up when renaming the variables after testing my PR. I prepped #12212 now which should correct that. Could you give that a whirl, please?

**Pull request merged**